lib/modal/add_user_modal.dart

```diff
@@ -195,67 +213,86 @@ Future addUserModal(
                   child: const Text('Cancel'),
                 ),
                 TextButton(
-                  onPressed: !_isFormValid
-                      ? null
-                      : () async {
-                          // add the product to the database
-                          final firstName = firstNameController.text;
-                          final lastName = lastNameController.text;
-                          final locationId =
-                              int.parse(locationIdController.text);
-                          final roleId = int.parse(roleIdController.text);
-                          final email = emailController.text;
-                          final phoneNumber = phoneNumberController.text;
-                          final password = getRandomString(20);
-
+                  onPressed: () async {
+                    _formKey.currentState!.validate();
```

Comment:
Should it continue to try and send a request even if the validator fails?

Reply:
Added if statement to check